macros/union_data.sql

```diff
@@ -22,67 +22,94 @@
     default_variable,
     union_schema_variable,
     union_database_variable
-    ) %}
+    ) -%}
 
-{% if var(union_schema_variable, none) %}
+{%- if var(union_schema_variable, none) -%}
 
-    {% set relations = [] %}
+    {%- set relations = [] -%}
     
-    {% if var(union_schema_variable) is string %}
-    {% set trimmed = var(union_schema_variable)|trim('[')|trim(']') %}
-    {% set schemas = trimmed.split(',')|map('trim'," ")|map('trim','"')|map('trim',"'") %}
-    {% else %}
-    {% set schemas = var(union_schema_variable) %}
-    {% endif %}
-
-    {% for schema in var(union_schema_variable) %}
-    {% set relation=adapter.get_relation(
+    {%- if var(union_schema_variable) is string -%}
+    {%- set trimmed = var(union_schema_variable)|trim('[')|trim(']') -%}
+    {%- set schemas = trimmed.split(',')|map('trim'," ")|map('trim','"')|map('trim',"'") -%}
+    {%- else -%}
+    {%- set schemas = var(union_schema_variable) -%}
+    {%- endif -%}
+
+    {%- for schema in var(union_schema_variable) -%}
+    {%- set relation=adapter.get_relation(
         database=source(schema, table_identifier).database if var('has_defined_sources', false) else var(database_variable, default_database),
         schema=source(schema, table_identifier).schema if var('has_defined_sources', false) else schema,
         identifier=source(schema, table_identifier).identifier if var('has_defined_sources', false) else table_identifier
     ) -%}
     
-    {% set relation_exists=relation is not none %}
+    {%- set relation_exists=relation is not none -%}
 
-    {% if relation_exists %}
+    {%- if relation_exists -%}
+        {%- do relations.append(relation) -%}
+    {%- endif -%}
 
-    {% do relations.append(relation) %}
+    {%- endfor -%}
     
-    {% endif %}
-
-    {% endfor %}
-
-    {{ dbt_utils.union_relations(relations) }}
-
-{% elif var(union_database_variable, none) %}
-
-    {% set relations = [] %}
-
-    {% for database in var(union_database_variable) %}
-
-    {% set relation=adapter.get_relation(
+    {%- if relations != [] -%}
+        {{ dbt_utils.union_relations(relations) }}
+    {%- else -%}
+    {% if execute and not var('remove_empty_table_warnings', false) -%}
```

Comment:
Could this maybe be misleading if customers believe this to be a native dbt variable? Should we consider renaming this to be `fivetran__remove_empty_table_warnings` or something like that?

Reply:
yeah i was thinking of a fivetran prefix as well! i'll update

Reply:
ah now i remember -- i opted for no prefix at first to be consistent with the `has_defined_sources` variable we also reference in this macro (IMO, _that_ name is confusing and should have a `fivetran__` prefix as well, but i don't wanna break anything here 😬)

Reply:
Yeah let's flag this as something to update in the next breaking version of fivetran_utils. I like the update!